src/frontend/README.md

```diff
@@ -1,7 +1,121 @@
-# frontend
+# frontend service
 
-Run the following command to restore dependencies to `vendor/` directory:
+The **frontend** service is responsible for rendering the UI for the store's website.
+It serves as the main entry point for the application routing requests to their 
+appropriate backend service.
+The application uses Server Side Rendering (SSR) to generate HTML consumed by 
+the browser.
 
-```sh
-dep ensure --vendor-only
+The following routes are defined by the frontend:
+
+| Path              | Method | Use                               |
+|-------------------|--------|-----------------------------------|
+| `/`               | GET    | Main index page                   |
+| `/cart`           | GET    | View Cart                         |
+| `/cart`           | POST   | Add to Cart                       |
+| `/cart/checktout` | POST   | Place Order                       |
+| `/cart/empty`     | POST   | Empty Cart                        |
+| `/logout`         | GET    | Logout                            |
+| `/product/{id}`   | GET    | View Product                      |
+| `/setCurrency`    | POST   | Set Currency                      |
+| `/static/`        | *      | Static resources                  |
+| `/robots.txt`     | *      | Search engine response (disallow) |
+| `/_healthz`       | *      | Health check (ok)                 |
+
+## OpenTelemetry instrumentation
+
+### Initialization
+
+The OpenTelemetry SDK is initialized in `main` using the `InitTraceProvider` function.
+
+```go
+func InitTracerProvider() *sdktrace.TracerProvider {
+    ctx := context.Background()
+
+    exporter, err := otlptracegrpc.New(ctx)
+    if err != nil {
+        log.Fatal(err)
+    }
+    tp := sdktrace.NewTracerProvider(
+        sdktrace.WithSampler(sdktrace.AlwaysSample()),
```

Comment:
Should we specify the always on sampler as a service requirement? Glancing through the other services it looks inconsistent today

Reply:
It seems like all the go-based services specify AlwaysSample, but the others do not.  The default is to AlwaysSample. Perhaps we should remove this from the go-services to avoid confusion for now, until we come up with a better plan on how to approach head-based sampling in this app?

Reply:
Yeah lets put together a comprehensive sampling plan. I'll open a separate pr for the other go apps

Reply:
https://github.com/open-telemetry/opentelemetry-demo-webstore/issues/101